### DISCUSSION TITLE
Awk tutorial

### DISCUSSION THREAD

**Discussion by code-chai** [sandbox-bio/sandbox.bio#17] [Q&A] [answered]
Hi! Thank you for putting together this fantastic introduction. For the last challenge on the tutorial, I feel like I'm close, but the sandbox is unsatisfied.
My attempt:
```
awk -F "\t" '\
{if($3 ~/Burrito/ && $4 ~/Guacamole/) counts[$3] += 1;\
else denom += 1}\
END {for(k in counts) print(k, counts[k]/denom)}' orders.tsv > burritos_guac.tsv
```
I'm sure I've made a silly mistake somewhere, but any help would be appreciated.

**Comment by robertaboukhalil** [comment, marked as answer]
Hi @code-chai,

Thanks for the kind words! This exercise is a little tricky (and probably not a great fit for an awk one-liner haha)

If I modify your code to output the denominator, I see it's always `4234`, whereas we want the denominator to be different for every burrito, since we want to know the percentage of burritos of each type that contain guac (i.e. `denom` will also need to be an array).

```bash
awk -F "\t" '\
{if($3 ~/Burrito/ && $4 ~/Guacamole/) counts[$3] += 1;\
else denom += 1}\
END {for(k in counts) print(k, denom)}' orders.tsv

Steak Burrito 4234
Veggie Burrito 4234
Barbacoa Burrito 4234
Chicken Burrito 4234
Burrito 4234
Carnitas Burrito 4234
```

Also, you'll want to always increment the denominator and not have it under an `else` statement.

Hope this helps!

**Comment by deve105** [comment]
after one week, I think i got it,

awk -F "\t" '{if($3 ~/Burrito/ &&  $4 ~/Guacamole/) countsa[$3]+=$2; if($3 ~/Burrito/) countsb[$3]+=$2} END {for (k in countsa) print (k, countsa[k]/countsb[k])}' orders.tsv > burritos_guac.tsv

**Reply by cnzed99** [reply]
this is my answer，awk -F "\t" '
{ 
  if ($3 ~ /Burrito/) all_type_count[$3] += $2
  if ($3 ~ /Burrito/ && $4 ~ /Guacamole/) count[$3] += $2
}
END { 
	for (k in count) print k, count[k]
	for (m in all_type_count) print m , all_type_count[m]
	
}' orders.tsv > burritos_guac.tsv
